tests/storages_tests/test_storages.py

```diff
@@ -579,9 +579,6 @@ def test_set_trial_param(storage_mode: str) -> None:
         # Check the previous call does not change the params.
         with pytest.raises(KeyError):
             storage.get_trial_param(trial_id_2, "y")
-        # State should be checked prior to distribution compatibility.
-        with pytest.raises(UpdateFinishedTrialError):
-            storage.set_trial_param(trial_id_2, "y", 0.4, distribution_z)
```

Comment:
This PR changes the order of checks for state and distribution compatibility, causing `ValueError` exceptions to be thrown. Since it's not critical whether `UpdateFinishedTrialError` or `ValueError` is raised in this test case, I'll remove the test.